lib/takeaway.rb

```diff
@@ -0,0 +1,52 @@
+require './lib/restaurant'
+
+class TakeAway
+  attr_reader :restaurant, :dish, :basket
+
+  def initialize(restaurant = Restaurant.new)
+    @restaurant = restaurant
+    @dish = ""
+    @basket = Hash.new(0)
+  end
+
+  def read_menu
+    restaurant.read_menu
+  end
+
+  def order(dish, qty=1)
+    fail "This dish is not available on menu" if not_on_menu?(dish)
+    basket[dish] += qty
+    "#{qty}x #{dish}(s) added to your basket."
+  end
+
+  def basket_summary
+    fail "basket is empty, no dishes ordered" if basket.empty?
+    my_basket
+  end
+
+  def order_value
+    "Total: "+ sprintf('£%.2f', total)
+  end
+
+  def confirm_order(amount=0)
+    fail "There are no dishes ordered, please order some first" if basket.empty?
+    fail "£#{sprintf('%.2f', amount)} does not match order value of #{sprintf('£%.2f', total)}" if amount != total
```

Comment:
Metrics/LineLength: Line is too long. [114/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)<br>Style/FormatString: Favor `format` over `sprintf`. (https://github.com/bbatsov/ruby-style-guide#sprintf)